File: Programowanie_Obiektowe/PO_C/PO_C/Guarana.cpp
#include "Guarana.h"

Guarana::Guarana(int x, int y, Swiat& swiat) : Roslina(GUARANA_SILA, x, y, swiat) {}

void Guarana::kolizja(Organizm& atakujacy)
{
    swiat.setLog(swiat.getLog() + string(1, rysowanie()) + " (" + to_string(x) + ", " + to_string(y) + ") zjedzony przez " + string(1, atakujacy.rysowanie()) + " (" + to_string(atakujacy.getX()) + ", " + to_string(atakujacy.getY()) + ")" + "\n");

    atakujacy.setSila(atakujacy.getSila() + GUARANA_BOOST_SILA);

    swiat.ustawOrganizmNa(atakujacy.getY(), atakujacy.getX(), nullptr);
    atakujacy.setX(x);
    atakujacy.setY(y);
    swiat.ustawOrganizmNa(atakujacy.getY(), atakujacy.getX(), &atakujacy);

    swiat.usunOrganizm(this);
}

char Guarana::rysowanie() const
{
    return GUARANA;
}

Organizm* Guarana::stworzPotomka(int x, int y) const
{
    return new Guarana(x, y, swiat);
}

File: Programowanie_Obiektowe/PO_JAVA/src/Guarana.java
public class Guarana extends Roslina
{
    public Guarana(int x, int y, Swiat swiat)
    {
        super(Stale.GUARANA_SILA, x, y, swiat);
    }

    @Override
    public void kolizja(Organizm atakujacy)
    {
        swiat.dodajLog(rysowanie() + " (" + x + ", " + y + ") zjedzony przez " + atakujacy.rysowanie() + " (" + atakujacy.getX() + ", " + atakujacy.getY() + ")");

        atakujacy.setSila(atakujacy.getSila() + Stale.GUARANA_BOOST_SILA);

        swiat.ustawOrganizmNa(atakujacy.getX(), atakujacy.getY(), null);
        atakujacy.setX(x);
        atakujacy.setY(y);
        swiat.ustawOrganizmNa(x, y, atakujacy);

        swiat.usunOrganizm(this);
    }

    @Override
    public char rysowanie()
    {
        return Stale.GUARANA;
    }

    @Override
    public String nazwa()
    {
        return Stale.GUARANA_S;
    }

    @Override
    public Organizm stworzPotomka(int x, int y)
    {
        return new Guarana(x, y, swiat);
    }
}